test/suite/test_checkpoint33.py

```diff
@@ -0,0 +1,126 @@
+from suite_subprocess import suite_subprocess
+import time
+from wiredtiger import stat
+from wtdataset import SimpleDataSet
+
+# test_checkpoint33.py
+#
+# Test that checkpoint will not skip tables that have available space at the end that can be
+# reclaimed through truncation.
+class test_checkpoint33(test_cc_base, suite_subprocess):
+    create_params = 'key_format=i,value_format=S,allocation_size=4KB,leaf_page_max=32KB,'
+    uri = 'table:test_checkpoint33'
+
+    table_numkv = 10000
+    value_size = 1024
+    value = 'a' * value_size
+
+    def delete(self, timestamp):
+        c = self.session.open_cursor(self.uri, None)
+        for k in range(self.table_numkv):
+            c.set_key(k)
+            self.session.begin_transaction()
+            c.remove()
+            self.session.commit_transaction("commit_timestamp=" + self.timestamp_str(timestamp))
+        c.close()
+
+    def populate(self, timestamp):
+        c = self.session.open_cursor(self.uri, None)
+        for k in range(self.table_numkv):
+            self.session.begin_transaction()
+            c[k] = self.value
+            self.session.commit_transaction("commit_timestamp=" + self.timestamp_str(timestamp))
+        c.close()
+
+    def get_size(self):
+        c = self.session.open_cursor('statistics:' + self.uri, None, None)
+        file_size = c[stat.dsrc.block_size][2]
+        c.close()
+        return file_size
+
+    def evict_all(self):
+        evict_cursor = self.session.open_cursor(self.uri, None, "debug=(release_evict)")
+        self.session.begin_transaction()
+        for k in range(self.table_numkv):
+            evict_cursor.set_key(k)
+            evict_cursor.search()
+            evict_cursor.reset()
+        self.session.rollback_transaction()
+        evict_cursor.close()
+
+    def test_checkpoint33(self):
+        # Pin oldest timestamp 1.
+        self.conn.set_timestamp(f'oldest_timestamp={self.timestamp_str(1)}')
+
+        # Create and populate a table at timestamp 2.
+        self.session.create(self.uri, self.create_params)
+        self.session.checkpoint()
+        self.populate(timestamp=2)
+
+        # Make everything stable at timestamp 3.
+        self.conn.set_timestamp(f'stable_timestamp={self.timestamp_str(3)}')
+
+        # Write to disk.
+        self.session.checkpoint()
+
+        # Delete everything at timestamp 4.
+        self.delete(timestamp=4)
+
+        # Make the deletions stable at timestamp 5.
+        self.conn.set_timestamp(f'stable_timestamp={self.timestamp_str(5)}')
+
+        # Write to disk.
+        self.session.checkpoint()
+
+        # Evict all pages
+        self.evict_all()
+
+        # Make everything globally visible.
+        self.conn.set_timestamp(f'oldest_timestamp={self.timestamp_str(5)}')
+        self.prout(f'File size: {self.get_size()}')
+
+        # Give checkpoint cleanup enough opportunity to clean up all the deleted pages.
+        cc_success = 0
+        while (cc_success < 1):
```

Comment:
`while` loop is no longer needed if we do it once.

Reply:
The while loop is to wait for the checkpoint cleanup to finish.

Reply:
Oh my bad, I misread. Should we use `wait_for_cc_to_run` instead? I think I raised this comment before.

Reply:
Yes, it can be used, that unify the logic everywhere in the tests to wait for the checkpoint cleanup.

Reply:
Fixed https://github.com/wiredtiger/wiredtiger/pull/11176/commits/c36ea30c281a2148b2373942897be01d842bd8d5